pkg/pillar/cmd/zedrouter/dnsmasq.go

```diff
@@ -140,61 +141,97 @@ func createDnsmasqConfiglet(
 	dhcphostsDir := dnsmasqDhcpHostDir(bridgeName)
 	ensureDir(dhcphostsDir)
 
-	file.WriteString(dnsmasqStatic)
+	var b bytes.Buffer
+
+	createDnsmasqConfigletNoFile(&b, ctx, bridgeName, bridgeIPAddr, netstatus, hostsDir, ipsetHosts, uplink, dnsServers, ntpServers)
+
+	file.WriteString(b.String())
+}
+
+func dhcpv4RangeConfig(start, end net.IP) (string, error) {
+	var dhcpRange string
+
+	if start != nil {
+		if end == nil {
+			dhcpRange = fmt.Sprintf("%s,static", start.String())
+		} else {
+			if bytes.Compare(start, end) > 0 {
+				return dhcpRange, fmt.Errorf("createDnsmasqConfigletWithBuffer: invalid dhcp-range %s-%s",
+					start.String(), end.String())
+			}
+			// full dhcp-range is not possible with static, see https://thekelleys.org.uk/dnsmasq/docs/dnsmasq-man.html
+			// "dhcp-range=[tag:<tag>[,tag:<tag>],][set:<tag>,]<start-addr>[,<end-addr>|<mode>[,<netmask>[,<broadcast>]]][,<lease time>]"
+			dhcpRange = fmt.Sprintf("%s,%s", start.String(), end.String())
+		}
+	}
+
+	return dhcpRange, nil
+}
+
+func createDnsmasqConfigletNoFile(
```

Comment:
What about renaming to something like `writeDnsmasqConfigletToBuf`. I just don't really like `NoFile` suffix. It sounds like it is not actually suitable for creating a config file.